application/image_processor.py

```diff
@@ -1,46 +1,223 @@
 
-        # raw_image = Image.open('C:/Users/ianth/Downloads/coca-cola-ad.jpg').convert("RGB")
+    def process_input(self) -> None:
+        """
+        This method will use the user's input to call the pre-trained model and provide the output.
+        :return: None
+        """
 
         processor = BlipProcessor.from_pretrained("Salesforce/blip-image-captioning-large")
         model = BlipForConditionalGeneration.from_pretrained("Salesforce/blip-image-captioning-large")
 
-        # Conditional image captioning
-        text = "a photography of"
-        inputs = processor(img, text, return_tensors="pt")
+        # Receive the given input as a Pillow Image object and the given source
+        images: list[tuple[Image, str]] | None = self.convert_input()
+
+        try:
+            # clean the input of any bad files
+            images = prune(images)
+        except TypeError:
+            self.invalid_prompt()
+
+        print('\nProcessing...')
+
+        for img, img_source in images:
+            print()  # to space the captions generated in the terminal
+
+            caption_image(img, img_source, processor, model)
+            break_line()
+
+    def is_url(self) -> bool:
+        """
+        This method checks if the user's input contains a valid URL(s). Many URL may be given at a time.
+        :return: True or False
+        """
+
+        self.urls: list[str] = self.user_input.split(', ')
+
+        valid: bool = True
+
+        for url in self.urls:
+            if not validators.url(url):
+                valid = False
+                break
+
+        return valid
+
+    def is_dir(self) -> bool:
+        """
+        This method checks if the user's input is an existing, local directory.
+        :return: True or False
+        """
+
+        return os.path.exists(self.user_input)
+
+    def clean_url(self) -> None:
+        """
+        If the given URL contains 'www,' add 'https://' to the beginning of the string so the model recognizes it.
+        :return: None
+        """
 
-        out = model.generate(**inputs)
-        print(processor.decode(out[0], skip_special_tokens=True))
+        if self.user_input[0:3] == 'www':
+            self.user_input = 'https://' + self.user_input
 
-        # Unconditional image captioning
-        inputs = processor(images=img, return_tensors="pt")
+    def invalid_prompt(self) -> None:
+        """
+        If the user input is completely invalid, prompt the user again to provide proper input.
+        :return:
+        """
+        utils.clear()
+        print(f'Invalid input, "{self.user_input}", was given. Please provide an image URL(s) or an existing directory '
+              f'to multiple images.')
+        self.user_input = input('\n> ')
 
-        out = model.generate(**inputs)
-        print(processor.decode(out[0], skip_special_tokens=True))
+        if not utils.quitting(self.user_input):
+            self.process_input()
```

Comment:
Pretty good documentation here. As someone who isn't the most experienced in Python, understood everything well